Rfc rc1

nmanthey opened the pull request:
*Issue #, if available:*

*Description of changes:*

 * disable gotocc as default
 * get rid of configuration directory
 * fix existing tests
 * add tests to include one-line-cr-bot
 * enable regression tests during CI

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

nmanthey committed:
markdown: adjust style in README.md and EXAMPLE.md

To make the files easier to process, adjust the style to what github is
used to.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
doc: add extra-clfag-set parameter

This is another useful feature that should be documented to be shared.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
doc: add description of one-line-cr-bot

Add a few lines about this script, and an example call to the README.md
file. Furthermore, add a more verbose example to the EXAMPLES.md file.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
doc: add one-line-cr-bot automation.

Make it easy for people to consume one-line-cr-bot as part of their
github control flow.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
Merge pull request #15 from nmanthey/README

Readme

nmanthey committed:
readme: fix link

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
Merge pull request #16 from nmanthey/README-fix-link

readme: fix link

nmanthey committed:
github: add comment script

This script allows to add a comment for a pull request, namely the pull
request that is currently running a workflow.

Authentication is done via the GITHUB_TOKEN environment variable, that
should be forwarded from github secrets.

The comments to be posted should come from a findings file. The user can
choose to only comment a summary.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
one-line-cr-bot: support commenting

We can comment on PRs and have a script added to the repository. This
changes makes this feeature available to consumers of one-line-cr-bot.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
one-line-cr-bot: fix spelling ignore

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
ols: do not log env

There might be secrets stored in the environment. To not leak them, do
not store the full environment in logs.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

MegaMaddin committed:
docker: add a minimum dockerfile

This adds a Dockerfile which can be used as a base image in other projects to
support one-line-scan in a docker container.

A sample invocation to build the "one-line-scan" image, could be:

```
docker build -t one-line-scan tools/
```

Later on, this can be used in other projects as a base image.

Signed-off-by: Martin Mazein <amazein@amazon.de>

nmanthey committed:
Merge pull request #17 from MegaMaddin/dockerfile

docker: add a minimum dockerfile

nmanthey committed:
comment-pr: skip reports for no defects

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
cr-bot: succeed after commenting

When we are posting comments to a PR successfully, we might not want to
block automation further. Hence, allow users to configure this
appropriately.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
cr-bot: do not append output file

In case an existing output file is used, we never truncate this file.
Hence, using the same file multiple times might result in misleading
output. Hence, do not append to this file.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
cr-bot: respect enabled tools

One install tools and present their version information in case the
tool is actually enabled. This avoids confusing output, as well as
avoids setup time for tools we will not use.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
github: do not fail on push

In case this script is invoked outside a pull request, the event
structure is different, and there will not be a pull_request member. As
this script might be used in automation for e.g. pushes, this use case
should be supported as well. Hence, this change implements checking for
the presence of the member first, and handling the cases accordingly.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

MegaMaddin committed:
one-line-cr-bot: show errors for git commands

Instead of throwing every output away, show errors during git execution for
easier debugging.

Signed-off-by: Martin Mazein <amazein@amazon.de>

nmanthey committed:
Merge pull request #18 from MegaMaddin/verbosity

one-line-cr-bot: show errors for git commands

nmanthey committed:
rc: drop configuration path

This directory has been here for historical reasons. As it is not
needed, drop it and update all relevant locations.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
cppcheck: fix test and flags

This change fixes the test for cppcheck, as well as fixes a bug
introduced here.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
tests: skip fortify test

In case the target test system does not provide Fortify, we cannot run
the test to test its features.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
afl: fix logging

AFL logging was currently done to the wrong path. This change fixes
logging for the AFL backend.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
AFL: do not export AFL_PATH

Using this environment variable might cause problems with the setup.
In case compilation fails with afl, you can set this environment
variable externally. For the systems afl 2.52, this variable needs
to be dropped.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
tests: fix AFL

When testing for AFL, we need to make sure we can re-use an existing
directory. Otherwise, re-running the used steps to reproduce them
is challenging.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
setup: make cbmc conditional

This change prepares to disable CBMC as the default backend.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
setup: count injected backends

Count injected backends and present this number during setup.

This change prepared disabling cbmc as the default tool.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
cbmc: introduce parameter to activate

To be able to disable CBMC as the default, we need a parameter that
allows to explicitly activate the CBMC backend during setup.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
cbmc: disable as default

This change disables CBMC as the default backend.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
no-gotocc: drop usage

As we removed CBMC as the default backend, also drop the parameter that
would disable this backend in the usage examples, as well as test runs
and consumers of the one-line-scan script.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
tests: add one-line-cr-bot test

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
tests: always run regression tests

We want to make sure to not break use cases. Hence, always run
regression tests.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
cbmc: do not require goto-diff

The tool goto-diff is not installed in PATH when the cbmc package is
instally via the system. Hence, do not rely on this tool being present.
However, if it is present, use it as before this change.

By this, we basically lose the recursive include analysis for the CBMC
backend.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
cbmc: use goto-cc  as goto-ld

Wen using CBMC from a system package, goto-ld is not present. To be
able to still use the CBMC backend. Use goto-gcc as tool for goto-ld.
However, create a link with the proper name, so that the tool itself
notices that is should use the ld CLI interface.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
test: fix cbmc test

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
tests: skip cbmc if not available

In case there is no CBMC available, do not test for relevant features.
This mainly addresses the version difference between the developed tool
and the currently available package in the CI system. Therefore, CBMC is
also not used as part of CI testing.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

nmanthey committed:
doc: improve one-line-cr-bot

The bot got more features, essentially it can comment on the github PR
communication stream by now. As that is a better UX, also mention this
as the default use case in the documentation.

Signed-off-by: Norbert Manthey <nmanthey@amazon.de>

tautschnig commented on backends/cbmc/gotocc-wrapper.sh:
```diff
@@ -258,13 +258,16 @@ function checkIncludes
                   sed -e 's/^.*#include[[:space:]]*[<"]//' \
                       -e 's/[>"][[:space:]]*$//')
-      if $GOTO_DIFF_BINARY -u "$tmpfile.o" "${tmpfile}-i.o" | grep -q "^[+-]"
+      if [ "$GOTO_DIFF_BINARY" != "$XX/usr/bin/goto-diffXX" ]
```
Spurious `$` before `XX/usr/bin/goto-diffXX`

nmanthey replied:
True, will fix.

tautschnig reviewed (approved):
Looks good to me except for one spurious `$` to be removed.

nmanthey commented:
Main diff of the update: rebased on top of the other PR, fixed above nit. Fixed another issue related to goto-ld handling when using the CBMC package that does not bring goto-ld. Added documentation update.
